### PR TITLE
Primitive arithmetic macro

### PR DESCRIPTION
**Pull request by elferherrera**
Definition of macro for primitive arithmetic matching. The macro is used to downcast `lhs` and `rhs` arrays to a primitive array type and executed the primitive arithmetic function with the results.

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -20,6 +20,8 @@ pub enum ArrowError {
     /// Error during import or export to/from a format
     ExternalFormat(String),
     DictionaryKeyOverflowError,
+    /// Error produced when unable to downcast an array
+    DowncastError(String),
```
Comment (jorgecarleitao) [inline src/error.rs:24]: Note that a downcast error never happens by design: we validate that an array has the right `DataType` on all constructors, so the error will appear before (when the user tries to initialize an array with a DataType that is invalid). This is different from `arrow`, where no such validation is currently made.

My suggestion is that we keep these as panics, as they represent a development error on our side. there is also nothing that the users of this library can do much about...

**Review by jorgecarleitao — COMMENTED** [review]
Great stuff. Lots of less code! Thanks a lot! :)

Let a comment that IMO we should address. Other than that, ready to :shipit: :)

File: src/error.rs (hunk 1 above)
Comment (elferherrera) [reply src/error.rs:24]: I see. It makes sense. Let me change them to panics. Do you think we should keep the DowncastError in the Enum or it should be removed as well?

Comment (jorgecarleitao) [reply src/error.rs:24]: Let's remove them: this way it is clear to end users that that is not an error that we emit.

**Pull request merged**